Ally_Desktop_Assistant.py

```diff
@@ -0,0 +1,464 @@
+from tkinter import *
+import tkinter.messagebox as tmsg
+from tkinter.filedialog import askdirectory
+import pyttsx3
+import speech_recognition as sr
+import time
+import webbrowser
+import json
+import os
+import random
+
+name = "Ally"
+name = name.title()
+
+data_filename = "Ally_Data.json"
+history_filename = "Ally_History.txt"
+tut_address = r"https://www.youtube.com/watch?v=v6YqIklzYZU"
+
+
+def show(data):
+    time_to_show = time.strftime("(%I:%M %p)")
+    chat_history.configure(state=NORMAL)
+    chat_history.insert(END, f"{time_to_show} {data}\n")
+    chat_history.yview_pickplace(END)
+    chat_history.update()
+    chat_history.configure(state=DISABLED)
+    with open(history_filename, "a") as f_obj:
+        f_obj.write(f"{time.ctime()}  {data}\n")
+
+
+def time_am_pm():
+
+    pr_time = time.strftime('%I:%M %p')
+    return pr_time
+
+
+def get_date():
```

Comment:
Misleading name, you are not returning a date object. You are returning a sentence containing formated date.